packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -143,9 +145,9 @@ async function fetchLatestSnapshotCore(
     storageTokenFetcher: (options: TokenFetchOptions, name: string) => Promise<string | null>,
     snapshotOptions: ISnapshotOptions | undefined,
     logger: ITelemetryLogger,
-    snapshotDownloader: (url: string, fetchOptions: {[index: string]: any}) => Promise<IOdspResponse<IOdspSnapshot>>,
+    snapshotDownloader: <T>(url: string, fetchOptions: {[index: string]: any}) => Promise<IOdspResponse<T>>,
```

Comment:
I'm a bit confused about usage of T here.
I'd expect compiler to require it to be generics arg for a function (fetchLatestSnapshotCore)
But also usage below assumes only IOdspSnapshot, so should T be removed (at a generic argument) and replaces (return value) with IOdspSnapshot?

Reply:
Here snapshotDownloader function is a generic which can return different types based on T like IOdspSnapshot or the newer format. It does not need to respond with IOdspsnapshot as the new format is coming. So the caller will know what type of result will it be.

Reply:
Let's assume that we now have two formats. How this function (fetchSnapshotWithRedeem) knows what format it gets from downloader? And why responsibility of conversion is not pushed to downloader (or even one layer below)?

Reply:
fetchLatestSnapshotCore only know which type of snapshot needs to fetched as that creates url and based on that it can call different converters to convert into common format.

Reply:
That makes sense. But I think we need to chose between the two options:
1. snapshotDownloader is some basic wrapper around fetch() that does not really know type it returns. It should be marked as returning `unknown` and the caller should do appropriate casting based on its knowledge of what it works with.
2. we expect callback to return IOdspSnapshot, as that's what this function expects.

I think # 2 is more suiting here - it's a requirement of this function to have a callback that returns IOdspSnapshot. How it's implemented / provided (and does it simply cases unknown to IOdspSnapshot) - it does not really matter from POV of this function.

Reply:
Requirement to get IOdspSnapshot if for older format. Once we integrate new format, then snapshotDownloader will not return IOdspSnapshot rather will return the newer format. So that is why I made snapshotDownloader generic and it does not need to know what type is being fetched.

Reply:
Only the fetchLatestSnapshotCore will know what type is being fetched.